hail/python/hailtop/aiotools/fs.py

```diff
@@ -583,6 +686,14 @@ async def close(self) -> None:
         for fs in self._filesystems:
             await fs.close()
 
-    async def copy(self, transfer: Union[Transfer, List[Transfer]]):
-        copier = Copier(self)
-        await copier.copy(transfer)
+    async def copy(self, transfer: Union[Transfer, List[Transfer]], raise_first_exception=True):
+        worker_pool = AsyncWorkerPool(50)
+        try:
+            copier = Copier(self)
+            copy_report = CopyReport(transfer)
```

Comment:
Did you test the copy report by hand?

Reply:
Not yet.  It is currently untested, except that the first exception is the one we expected.